SingleSource/UnitTests/AArch64/acle-function-multi-versioning.c

```diff
@@ -0,0 +1,288 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#ifdef __APPLE__
+#include <sys/sysctl.h>
+#endif
+
+static bool safe_try_feature(bool (*try_feature)(void));
+
+static bool any_fails = false;
+
+#if __HAVE_FUNCTION_MULTI_VERSIONING
+#define CHECK(X, BODY) \
+    __attribute__((target(#X))) \
+    static bool try_##X(void) { \
+        do \
+            BODY \
+        while (0); \
+        return true; \
+    } \
+    __attribute__((target_version(#X))) \
+    static void check_##X(void) { \
+        printf("%s\n", #X); \
+        fflush(stdout); \
+        if (!safe_try_feature(try_##X)) { \
+            printf("\tFAIL\n"); \
+            any_fails = true; \
+        } \
+    } \
+    __attribute__((target_version("default"))) \
+    static void check_##X(void) { \
+        printf("%s\n", #X); \
+        fflush(stdout); \
+        if (safe_try_feature(try_##X)) { \
+            printf("\tUPASS\n"); \
+            any_fails = true; \
+        } \
+    }
+#else
+#define CHECK(X, BODY) \
+    static void check_##X(void) { \
+        printf("%s\n", #X); \
+    }
+#endif
+
+CHECK(flagm, {
+    asm volatile (
+        "cfinv" "\n"
+        "cfinv" "\n"
+    );
+})
+CHECK(flagm2, {
+    asm volatile (
+        "axflag" "\n"
+        "xaflag" "\n"
+    );
+})
+CHECK(dotprod, {
+    asm volatile (
+        "udot v0.4S,v1.16B,v2.16B"
+        : : : "v0"
+    );
+})
+CHECK(sha3, {
+    asm volatile (
+        "fmov d0, #0" "\n"
+        "fmov d1, #0" "\n"
+        "eor3 v0.16b, v0.16b, v0.16b, v0.16b" "\n"
+        : : : "v0"
+    );
+})
+CHECK(rdm, {
+    asm volatile (
+        "sqrdmlah s0, s1, s2"
+        : : : "s0"
+    );
+})
+CHECK(lse, {
+    uint64_t pointee = 0;
+    asm volatile (
+        "swp xzr, xzr, [%[pointee]]"
+        : : [pointee]"r"(&pointee)
+    );
+})
+CHECK(sha2, {
+    asm volatile (
+        "fmov d0, #0" "\n"
+        "fmov d1, #0" "\n"
+        "sha256h q0, q0, v0.4s" "\n"
+        : : : "v0"
+    );
+})
+CHECK(sha1, {
+    asm volatile (
+        "fmov s0, #0" "\n"
+        "sha1h s0, s0" "\n"
+        : : : "v0"
+    );
+})
+CHECK(aes, {
+    asm volatile (
+        "fmov d0, #0" "\n"
+        "fmov d1, #0" "\n"
+        "aesd v0.16B, v0.16B" "\n"
+        : : : "v0"
+    );
+})
+CHECK(pmull, {
+    asm volatile (
+        "fmov d0, #0" "\n"
+        "pmull v0.1q, v0.1d, v0.1d" "\n"
+        : : : "v0"
+    );
+})
+CHECK(rcpc, {
+    int x;
+    asm volatile (
+        "ldaprb w0, [%0]"
+        : : "r" (&x) : "w0"
+    );
+})
+CHECK(rcpc2, {
+    int x;
+    asm volatile (
+        "ldapurb w0, [%0]"
```

Comment:
The same as above, rcpc2 mapped to +rcpc and the asm check will complain on rcpc2 instruction.